### PR TITLE
Autogenerated API types

### PR DESCRIPTION
**Pull request by Jeffrey-Zutt**
- Added a type-generator to generate types based on swagger configuration.
- Replaced `Settings`-related types with the auto-generated ones.

NOTE:
As the autogenerated-types are not part of the repository, I don't think the build still succeeds.
Should we add the `swagger:generate-schema` to the build-steps?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +61,8 @@
     "lint:quiet": "npm run lint -- --quiet",
     "eject": "react-scripts eject",
     "storybook": "start-storybook -p 9009 -s public",
-    "build-storybook": "build-storybook -s public"
+    "build-storybook": "build-storybook -s public",
+    "swagger:generate-schema": "dtsgen -o ./src/__generated__/apiSchema.d.ts -n API --url https://acc.api.top.amsterdam.nl/api/v1/swagger/?format=openapi"
```
Comment (gilleswittenberg) [inline package.json:65]: We probably need:
`"swagger:generate-schema:acc": "dtsgen -o ./src/__generated__/apiSchema.d.ts -n API --url https://acc.api.top.amsterdam.nl/api/v1/swagger/?format=openapi"`
`"swagger:generate-schema": "dtsgen -o ./src/__generated__/apiSchema.d.ts -n API --url https://api.top.amsterdam.nl/api/v1/swagger/?format=openapi"`

Comment (gilleswittenberg) [inline package.json:65]: Also this needs to be added as a build step in Jenkins. And README for development.

File: package.json [hunk 2]
```diff
@@ -60,7 +61,8 @@
     "lint:quiet": "npm run lint -- --quiet",
     "eject": "react-scripts eject",
     "storybook": "start-storybook -p 9009 -s public",
-    "build-storybook": "build-storybook -s public"
+    "build-storybook": "build-storybook -s public",
```
Comment (gilleswittenberg) [inline package.json:64]: "storybook:build"?

File: package.json (hunk 1 above)
Comment (Jeffrey-Zutt) [reply package.json:65]: Good one, I'll add the other one as well.

File: package.json (hunk 2 above)
Comment (Jeffrey-Zutt) [reply package.json:64]: :+1:

File: package.json (hunk 1 above)
Comment (Jeffrey-Zutt) [reply package.json:65]: :+1:

**Pull request merged**